controllers/persistentvolumeclaim_controller.go

```diff
@@ -15,18 +16,31 @@ import (
 
 // PersistentVolumeClaimReconciler reconciles a PersistentVolumeClaim object
 type PersistentVolumeClaimReconciler struct {
-	client client.Client
+	client    client.Client
+	apiReader client.Reader
 }
 
 // NewPersistentVolumeClaimReconciler returns NodeReconciler.
-func NewPersistentVolumeClaimReconciler(client client.Client) *PersistentVolumeClaimReconciler {
-	return &PersistentVolumeClaimReconciler{client: client}
+func NewPersistentVolumeClaimReconciler(client client.Client, apiReader client.Reader) *PersistentVolumeClaimReconciler {
+	return &PersistentVolumeClaimReconciler{
+		client:    client,
+		apiReader: apiReader,
+	}
 }
 
 //+kubebuilder:rbac:groups=core,resources=persistentvolumeclaims,verbs=get;list;watch;update
 //+kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch;delete
 
 // Reconcile finalize PVC
+//
+// This is originally created as a workaround for the following issue.
+// https://github.com/kubernetes/kubernetes/pull/93457
+//
+// Because the issue was fixed, the PVC reconciler was once removed by PR #536.
```

Comment:
```
Because the issue was fixed, the PVC reconciler was once removed by PR #536.
However, it turned out that the PVC finalizer was also useful to resolve the issue that a pod created from StatefulSet persists in the PENDING state when PVC is deleted for some reasons like node deletion.
Thus, the reconciler was revived by PR #620.
```